rm-datasource/src/main/java/io/seata/rm/datasource/DataSourceProxy.java

```diff
@@ -94,11 +104,45 @@ public DataSourceProxy(DataSource targetDataSource, String resourceGroupId) {
         if (targetDataSource instanceof SeataDataSourceProxy) {
             LOGGER.info("Unwrap the target data source, because the type is: {}", targetDataSource.getClass().getName());
             targetDataSource = ((SeataDataSourceProxy) targetDataSource).getTargetDataSource();
+        } else {
+            initSeataDataSource(targetDataSource);
         }
         this.targetDataSource = targetDataSource;
         init(targetDataSource, resourceGroupId);
     }
 
+    private void initSeataDataSource(DataSource targetDataSource) {
+        String className = targetDataSource.getClass().getSimpleName();
+        if (className.startsWith("Druid") && targetDataSource instanceof DruidDataSource) {
```

Comment:
should not be directly depend on Druid or Hikari.

Reply:
> should not be directly depend on Druid or Hikari.

would first determine if the classname matches the criteria, perhaps it would be better to use reflection to determine if a Druid or Hikari dependency exists